crates/driver/src/domain/competition/auction.rs

```diff
@@ -457,12 +453,32 @@ impl AuctionProcessor {
         orders
     }
 
-    pub fn new(eth: &infra::Ethereum) -> Self {
+    pub fn new(eth: &infra::Ethereum, order_priority_config: OrderPriorityConfig) -> Self {
         let eth = eth.with_metric_label("auctionPreProcessing".into());
+        let order_timestamp_threshold =
+            Duration::from_std(order_priority_config.order_creation_timestamp_threshold).unwrap();
+        let mut order_comparators = Vec::<Arc<dyn sorting::OrderComparator>>::new();
+
+        order_comparators.push(Arc::new(sorting::OrderClass).comparator());
+
+        for strategy in order_priority_config.strategies {
+            let comparator: Arc<dyn sorting::OrderComparator> = match strategy {
+                OrderPriorityStrategy::ExternalPrice => {
+                    Arc::new(sorting::ExternalPrice).comparator()
```

Comment:
It is very strange to create an `Arc` of an object and then calling a function to it. Why not to call the function and then wrap it in an `Arc`?